sys-fs/xfsprogs/xfsprogs-9999.ebuild

```diff
@@ -0,0 +1,192 @@
+fi
+
+LICENSE="LGPL-2.1"
+SLOT="0"
+IUSE="icu libedit nls selinux static-libs"
+
+RDEPEND="
+	dev-libs/inih
+	dev-libs/userspace-rcu:=
+	>=sys-apps/util-linux-2.17.2
+	icu? ( dev-libs/icu:= )
+	libedit? ( dev-libs/libedit )
+"
+DEPEND="
+	${RDEPEND}
+	>=sys-kernel/linux-headers-6.11
+"
+BDEPEND="
+	${REGEN_BDEPEND}
+	nls? ( sys-devel/gettext )
+"
+RDEPEND+=" selinux? ( sec-policy/selinux-xfs )"
+
+call_prepare_release() {
+	local version
+
+	#test sed -i
+	#	's/^AC_INIT\(\[xfsprogs\],\[([0-9]+).([0-9]+).([0-9]+)\],\[linux-xfs@vger\.kernel\.org\]\)$/\1.\2.9a999/'
+	#	configure.ac
+
+	version="$(grep -o -- \
+		'^AC_INIT(\[xfsprogs\],\[[0-9]\+\.[0-9]\+\.[0-9]\+\],\[linux-xfs@vger.kernel.org\])$' \
+		configure.ac)"
+	if [ $? != 0 ]; then die; fi
+
+	#test version+='a'
+	version=$(echo "${version}" | ( sed -r \
+		's/^AC_INIT\(\[xfsprogs\],\[([0-9]+).([0-9]+).([0-9]+)\],\[linux-xfs@vger\.kernel\.org\]\)$/\1.\2.9999/;tm;${x;/1/{x;q};x;q1};b;:m;x;s/.*/1/;x' \
+		|| die ) )
+
+	#test sed -i 's/^update_version\(\) \{/aupdate_version()  {/g' release.sh
+	# https://stackoverflow.com/questions/8818119/how-can-i-run-a-function-from-a-script-in-command-line
+	local update_version_function=$(sed -n \
+		'/^update_version() {/,/^}$/p' release.sh) || die
+	update_version_function=$( echo "${update_version_function}" | (sed -r \
+		's/^update_version\(\) \{$//;tm;${x;/1/{x;q};x;q1};b;:m;x;s/.*/1/;x' \
+		|| die ) )
+	update_version_function=$( echo "${update_version_function}" | (sed -r \
+		's/^\}$//;tm;${x;/1/{x;q};x;q1};b;:m;x;s/.*/1/;x' \
+		|| die ) )
+	echo "${update_version_function}" > ./release_update_version.sh
+
+	#test version+='a'
+	local version_file
+	# https://stackoverflow.com/questions/15965073/return-value-of-sed-for-no-match
+	local version_major=$( echo "${version}" | ( sed -r \
+		's/^([0-9]+).([0-9]+).([0-9]+)$/\1/;tm;${x;/1/{x;q};x;q1};b;:m;x;s/.*/1/;x' \
+		|| die ) )
+	version_file+="PKG_MAJOR="${version_major}""
+	version_file+=$'\n'
+	local version_minor=$( echo "${version}" | ( sed -r \
+		's/^([0-9]+).([0-9]+).([0-9]+)$/\2/;tm;${x;/1/{x;q};x;q1};b;:m;x;s/.*/1/;x' \
+		|| die ) )
+	version_file+="PKG_MINOR="${version_minor}""
+	version_file+=$'\n'
+	local version_revision=$( echo "${version}" | ( sed -r \
+		's/^([0-9]+).([0-9]+).([0-9]+)$/\3/;tm;${x;/1/{x;q};x;q1};b;:m;x;s/.*/1/;x' \
+		|| die ) )
+	version_file+="PKG_REVISION="${version_revision}""
+	version_file+=$'\n'
+	echo "${version_file}" > ./VERSION
+	chmod +x release_update_version.sh
+	# https://stackoverflow.com/questions/17583578/what-command-means-do-nothing-in-a-conditional-in-bash
+	EDITOR="true" version="${version}" ./release_update_version.sh || die
+}
+
+src_prepare() {
+	default
+
+	# Fix doc dir
+	sed -i \
+		-e "/^PKG_DOC_DIR/s:@pkg_name@:${PF}:" \
+		include/builddefs.in || die
+
+	# Don't install compressed docs
+	sed 's@\(CHANGES\)\.gz[[:space:]]@\1 @' -i doc/Makefile || die
+
+	if [[ ${PV} == *9999 ]]; then
+		call_prepare_release || die
```

Comment:
There's no point in the `die` here, as your function will die on its last command if it fails anyway (you have `... ./release_version_update.sh || die` and it doesn't ever early-return).